services/tables/src/test/java/com/linkedin/openhouse/tables/e2e/h2/ToggleH2StatusesRepository.java

```diff
@@ -11,4 +16,29 @@
  */
 @Repository
 @Primary
-public interface ToggleH2StatusesRepository extends ToggleStatusesRepository {}
+public interface ToggleH2StatusesRepository extends ToggleStatusesRepository {
+  String FEATURE_ACTIVATION_TABLE_NAME = "acv_tbl";
+  TblPropsToggleRegistry registry = new TblPropsToggleRegistry();
+
+  /**
+   * A trick to activate a specific feature for a specific table-id. This is only for testing
+   * purpose.
+   */
+  @Override
+  default Optional<TableToggleStatus> findById(ToggleStatusKey toggleStatusKey) {
```

Comment:
why this trick? couldn't we have autowired the repository and insertd the toggleKey before running the unit test. This trick will affect other test cases too?

Reply:
we don't support any write method for toggle endpoint. 

This is the same thing we do for mock test like below: 
```
public class MockTablesApiHandler implements TablesApiHandler {
  @Override
  public ApiResponse<GetTableResponseBody> getTable(
      String databaseId, String tableId, String actingPrincipal) {
    switch (databaseId) {
      case "d200":
        return ApiResponse.<GetTableResponseBody>builder()
            .httpStatus(HttpStatus.OK)
            .responseBody(RequestConstants.TEST_GET_TABLE_RESPONSE_BODY)
            .build();
      case "d404":
        throw new NoSuchUserTableException(databaseId, tableId);
      case "dnullpointer":
        throw new NullPointerException(); // test for exception handler
      default:
        return null;
    }
  }
```

Reply:
I see, the test you share is not e2e right? its `/mock`? should we try injecting a custom TestToggleH2StatusesRepository that lets us insert mock entries instead of hardcoding?

Reply:
yes this is from mock module. 
I just realize that the in-memory version of the repo can support `save` method for feature activation. Just use that to replace "trick" that was added in the earlier version.